### DISCUSSION TITLE
Experiencias de aprendizaje en proyecto

### DISCUSSION THREAD

**Discussion by kevinmamaqi** [IT-Academy-BCN/ita-wiki#1242] [Q&A]
Por favor, compartir vuestras experiencias de aprendizaje en el proyecto en esta discusión. Referencia de areas principales aquí: https://github.com/IT-Academy-BCN/ita-wiki/blob/main/docs/WHAT_YOU_LEARN.md

**Comment by marmorrei** [comment]
Mi experiencia en el proyecto complementa y amplía lo que hice durante el itinerario. Creo que para tener una formación completa es lo ideal, ya que aspectos como trabajar en un equipo con todo lo que esto conlleva (control de versiones, comunicación, diferentes especialidades, etc) no se pueden poner en práctica durante el itinerario y es lo primero que vas a tener que aprender cuando llegues a una empresa.
Por otro lado, en cuanto a nivel técnico noto que mi avance ha sido también muy significativo, ya que al ser un proyecto real, te pones las pilas sí o sí. Yo lo recomiendo 100% porque seguro que llegas a tu primer trabajo mucho más preparado.

**Comment by mllenas12** [comment]
Participar en este proyecto después de haber hecho el bootcamp en React me ha permitido poner en práctica todos los conocimientos adquiridos durante el curso, aprender muchos otros nuevos (storybook, tanstack query…) y reforzar los que menos segura me sentía (styled-components y, sobre todo, testing). Aunque lo que creo que más aporta esta fase de proyectos es lo que no puedes aprender en casa, como el trabajo en equipo, el trabajo en paralelo del equipo de frontend con el de backend, el control de versiones con Git, y entender un código que no has creado tú. Definitivamente lo recomendaría a cualquier persona que haya hecho el bootcamp, ya que estoy segura de que después de proyectos vas mucho más preparado a cualquier entrevista y, al empezar en una nueva posición de trabajo, será mucho más fácil, ya que proyectos es lo más parecido del bootcamp a estar trabajando en una empresa como desarrollador.

**Comment by MiguelKummetz** [comment]
Personalmente, creo que la experiencia de participar en este proyecto es muy valiosa. 
El aprendizaje durante el itinerario es importante, pero está muy limitado a capacidades técnicas individuales.
Para mí ha sido muy interesante el hecho de empezar a trabajar en equipo, en un proyecto que tiene un tamaño considerable y ya llevaba mucho recorrido y tener que ponerse al día, creo que es una experiencia muy valiosa de cara a entrar en una empresa.

**Comment by danny-mv** [comment]
Mi participación en este proyecto después de la realización del itinerario ha sido una experiencia bastante enriquecedora, permitiéndome aplicar los conocimientos adquiridos y profundizar en aspectos técnicos de la mano de un mentor con años de experiencia en el sector.
El trabajo colaborativo también es un aspecto clave; el realizar pair programming, participar en las revisiones de código y participar en el diseño e implementación de nuevas funcionalidades no solo han mejorado mi comunicación técnica, sino que me han proporcionado una forma para recibir y ofrecer retroalimentación constructiva haciendo que mejore la calidad del proyecto y la dinámica de los equipos.
Actualmente, [el mercado es muy competitivo](https://www.getmanfred.com/blog/guia-salarial-2024-salarios-en-tecnologia-espana-manfred) y adquirir la experiencia práctica ofrece la posibilidad de enfrentarte a las entrevistas con más confianza y con una perspectiva realista del trabajo en un entorno profesional.

**Comment by joanbr4** [comment]
En el poco timpo que llevo en el proyecto, con un recorrido previo al curso de Back-end de NodeJs, me ha permitido darme cuenta de lo poco que se y lo mal que estructuro mi codigo. Valga decir que esto es un proyecto grande y de equipo y no el que harias en tu casa. 
Desde el primero momento, me sentí abrumado por las mil capas de codigo existentes en las mil carpetas eso, bien organizadas haciendo uso de convenciones internacionales ( Google ),  pero el poder trabajar en equipo te permite empezar a pensar como la comunicacion es super importante, igual o mas que el código. 
La experiencia es hiper real cuando tienes que despertartre si o si y hacer las reuniones diarias que toda empresa tiene, te hace entender lo bueno y lo malo de las reuniones. 
Yo recomendaria a toda persona que se apunte al proyecto si quieren hacer una inmersion real y práctica, y que puedan aprovechar para hacer la transición a un entorno real simulado aquí, que hacerlo por primera vez en una empresa real.

**Comment by Xerxi88** [comment]
Mi experiencia al pasar a proyecto ha sido muy enriquecedora. Si que es verdad que al dar el salto la exigencia es exponencial y abrumadora ya que pasas de hacer pequeños proyectos para aprender a un proyecto grande y real que podría estar en cualquier empresa. Pero de esta pude desarrollar mucho mas mis conocimientos técnicos y mis habilidades de comunicación haciendo pair programing con los compañeros que siempre están dispuestos a ayudar ya que al fin y al cabo todos estamos en la misma situación de búsqueda de trabajo. Respecto a la búsqueda de trabajo no creo que estar en proyecto te reste tiempo para ello, sino mas bien al contrario, desarrolla mas tus capacidades como programador y te da mas soltura a la hora de pasar un proceso de selección enfrente de entrevistas técnicas por ejemplo.
Animo a todos los que acaben itinerario a unirse a proyecto y vivir la experiencia que a pesar de ser dura también es muy reconfortante.

**Comment by NataliaR-BCN** [comment]
Participar en el proyecto permite experimentar todo lo que significa integrarse en un equipo de desarrollo y contribuir a la definición y creación de aplicaciones junto con otros programadores, tanto de tu perfil/tecnología como de otras disciplinas.
Tu aprendizaje técnico se acelera con nuevos retos y herramientas, revisando el código de los compañeros, compartiendo ideas, aportando y recibiendo correcciones o sugerencias de mejora... 
Pero además de ayudarte a seguir evolucionando técnicamente, también sirve para aprender a desenvolverte en un entorno profesional, porque podrás ejercitar todas las habilidades que necesitarás en futuros empleos: trabajar con metodología ágil y en remoto, participar en las dailies y las retros, comunicarte con los compañeros y compartir objetivos, fusionar tu código con el de los demás, lidiar con la deuda técnica...
Puede parecer un salto respecto al itinerario pero en realidad es un excelente siguiente paso mientras buscas trabajo. Es una experiencia muy útil para practicar en un entorno seguro, coger rodaje para afrontar la etapa laboral con más seguridad y conseguir conocimientos que te ayudarán a adaptarte con más facilidad y rapidez a tu primer empleo como desarrollador.
